.travis.yml

```diff
@@ -29,97 +24,55 @@ env:
     - TF_VERSION_ID=  # Do not install TensorFlow in this case
 
 cache:
+  pip: true
```

Comment:
Okay, it looks like this just caches the Pip cache directory and not the
actual virtualenv contents, which is good. (If it cached the virtualenv,
we might be relatively safe anyway, given that all our Pip dependencies
are either pinned or installed with `-I`… but we could still be hit by
any Pip problems w.r.t. reinstalling packages in the same env.)

Reply:
Added a comment, but yes this is fine, though we do get weird pip cache deserialization errors fairly often for reasons that are unclear to me.

Reply:
ಠ_ಠ

:see_no_evil: 

lgtm

Reply:
The set of changes left after removing all of them from view trivially looks good, huh?